internal/util/search/search.go

```diff
@@ -0,0 +1,178 @@
+package search
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/pkg/kptfile"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const PathDelimiter = "."
+
+// SearchReplace struct holds the input parameters and results for
+// Search and Replace operations on resource configs
+type SearchReplace struct {
+	// Value is the value of the field to be matched
+	Value string
+
+	// ValueRegex is the value regex of the field to be matched
+	ValueRegex string
+
+	// Path is the path of the field to be matched
+	Path string
+
+	// Count is the number of matches
+	Count int
+
+	// Match is the map of file path to list of matched fields
+	Match map[string][]string
+
+	// PutLiteral is the literal to be put at to field
+	// filtered by path and/or value
+	PutLiteral string
+
+	filePath string
+}
+
+// Perform performs the search and replace operation on each node in the package path
+func (sr *SearchReplace) Perform(resourcesPath string) error {
+	inout := &kio.LocalPackageReadWriter{
+		PackagePath:     resourcesPath,
+		NoDeleteFiles:   true,
+		PackageFileName: kptfile.KptFileName,
```

Comment:
Not at all relevant to your PR, but why are we pluming this constant manually? It should be a global constant defined once (Doesn't mean it needs to be defined in upstream libraries).

Reply:
It is defined once and referenced multiple times. If this is not specified, it means reader should not honor package boundaries. Since reader is the core part for both kustomize and kpt, we need the upstream to mention the file name each time the LocalPackageReadWriter is invoked.